### DISCUSSION TITLE
basic-auth  插件无法关联路由或者服务吗？

### DISCUSSION THREAD

**Discussion by Lewisyixin** [apache/apisix#8056] [Q&A]
看文档描述以及自己测试的结果，
首先创建消费者，账号密码是在消费者里保存的，
其次路由和服务创建的时候只能选择开启basic-auth插件，但是无法关联某个消费者，
所以比如我路由a和路由b，想分别关联consumerA和consumerB（即各自使用各自的密码），有什么好办法实现吗？

**Comment by tzssangglass** [comment]
Are you saying that the route uses the consumer's password?

> （即各自使用各自的密码）

I didn't understand this. Can you be a little more detailed?

**Reply by Lewisyixin** [reply]
比如我昨天测试使用的结果是，首先创建两个consumer，c1和c2，c1和c2各自对应一组不同用户名和密码
![image](https://user-images.githubusercontent.com/17898688/194790780-76d37db8-4cc6-4221-ae53-e8865264e38e.png)
然后我希望两条不相关的路由routeA和routeB，开启密码，routeA使用c1的用户名密码，routeB使用c2的用户名密码，但是路由开启插件的时候无法关联consumer，只能选择开启basic-auth。
![image](https://user-images.githubusercontent.com/17898688/194790847-a7b051e3-71ec-45f2-b6d9-456a8974b534.png)

所以结果就是routeA和routeB都开启了插件之后，使用c1或c2均可以访问两个路由，我希望routeA只能使用c1，routeB只能使用c2

**Reply by tzssangglass** [reply]
It looks like you want to specify that the consumer accesses the specified route.

try with: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/consumer-restriction.md

**Reply by Lewisyixin** [reply]
It works, thx
